src/Internal/Service/OrderService.php

```diff
@@ -126,4 +142,105 @@ public function get_payment_metadata( int $order_id, Payment_Type $payment_type
 
 		return apply_filters( 'wcpay_metadata_from_order', $metadata, $order, $payment_type );
 	}
+
+	/**
+	 * Imports the data from an order to a payment context.
+	 *
+	 * @param int            $order_id ID of the order.
+	 * @param PaymentContext $context  A payment context, awaiting order data.
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function import_order_data_to_payment_context( int $order_id, PaymentContext $context ) {
+		$order = $this->get_order( $order_id );
+
+		$currency = strtolower( $order->get_currency() );
+		$amount   = WC_Payments_Utils::prepare_amount( $order->get_total(), $currency );
+
+		$user = $order->get_user();
+		if ( false === $user ) { // Default to the current user.
+			$user = $this->legacy_proxy->call_function( 'wp_get_current_user' );
+		}
+
+		$context->set_currency( $currency );
+		$context->set_amount( $amount );
+		$context->set_user_id( $user->ID );
+	}
+
+	/**
+	 * God method for updating orders once a payment has succeeded.
+	 *
+	 * @param int                                $order_id ID of the order that was just paid.
+	 * @param WC_Payments_API_Abstract_Intention $intent   Remote object. To be abstracted.
+	 * @param PaymentContext                     $context  Context for the payment.
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function update_order_from_successful_intent(
+		int $order_id,
+		WC_Payments_API_Abstract_Intention $intent,
+		PaymentContext $context
+	) {
+		$order = $this->get_order( $order_id );
```

Comment:
I have noticed that we keep getting order from DB. Can we store order in Payment Context and use it when needed?

Reply:
I've responded in a different PR: https://github.com/Automattic/woocommerce-payments/pull/7324#discussion_r1342773519